### PR TITLE
give-nft and give in wallet.hoon

### PR DESCRIPTION
**Pull request by marcoleder**
This PR changes a how give and give-nft are handled:

- %give and %give-nft were laid together: We introduce _amount-or-id_ which serves as a placeholder, and depending on whether we have %give or %give-nft will save the related information into it. Then we process as already given before.

### CODE REVIEW

File: app/wallet.hoon [hunk 1]
```diff
@@ -270,9 +270,14 @@
                         (fry-rice:smart `@ux`'zigs-contract' from.act town.act `@`'zigs')
       ::  need to check transaction type and collect rice based on it
       ::  only supporting small subset of contract calls, for tokens and NFTs
-      =/  formatted=[args=(unit *) our-grains=(set @ux) cont-grains=(set @ux)]
-        ?-    -.args.act
-            %give
+      =/  formatted=[args=(unit *) my-grains=(set @ux) cont-grains=(set @ux)]
```
Comment (dr-frmr) [inline app/wallet.hoon:273]: everything beneath the =/  formatted declaration is just one gap too indented

File: app/wallet.hoon [hunk 2]
```diff
@@ -270,9 +270,14 @@
...
+      ::  if sending NFT, save item-id
+          =/  amount-or-id
+            ?:  =(-.args.act %give-nft)  item-id.args.act
+            ::  else, save amount to send
+            ?:  =(-.args.act %give)  amount.args.act
+            !!
```
Comment (dr-frmr) [inline app/wallet.hoon:279]: can we add a `~|` error message for this crash?

**Review by dr-frmr — CHANGES_REQUESTED** [review]

File: app/wallet.hoon (hunk 1 above)
Comment (marcoleder) [reply app/wallet.hoon:273]: gotcha

File: app/wallet.hoon (hunk 2 above)
Comment (marcoleder) [reply app/wallet.hoon:279]: _in order to submit with %custom, use %submit-custom instead_

**Review by marcoleder — COMMENTED** [review]
see commit 795a10189995d4a7fa9947cdd973598ab5fa1682

**Review by dr-frmr — APPROVED** [review]

**Comment by dr-frmr** [comment]
looks good to me

**Pull request merged**